File: java/src/main/java/com/dokmatiq/docgen/model/InvoiceUnit.java
package com.dokmatiq.docgen.model;

import com.fasterxml.jackson.annotation.JsonValue;

/** Invoice unit codes (UN/ECE Recommendation 20). */
public enum InvoiceUnit {
    PIECE("C62"), HOUR("HUR"), DAY("DAY"),
    KILOGRAM("KGM"), METER("MTR"), LITER("LTR"),
    SQUARE_METER("MTK"), CUBIC_METER("MTQ"),
    SET("SET"), PACKAGE("PK");

    private final String value;

    InvoiceUnit(String value) { this.value = value; }

    @JsonValue
    public String getValue() { return value; }
}

File: php/src/Model/InvoiceUnit.php
<?php

declare(strict_types=1);

namespace Dokmatiq\DocGen\Model;

/** Invoice unit codes (UN/ECE Recommendation 20). */
enum InvoiceUnit: string
{
    case PIECE = 'C62';
    case HOUR = 'HUR';
    case DAY = 'DAY';
    case KILOGRAM = 'KGM';
    case METER = 'MTR';
    case LITER = 'LTR';
    case SQUARE_METER = 'MTK';
    case CUBIC_METER = 'MTQ';
    case SET = 'SET';
    case PACKAGE = 'PK';
}
